src/legacy/core_plugins/metrics/public/components/aggs/percentile_rank.js

```diff
@@ -19,33 +19,91 @@
 
 import PropTypes from 'prop-types';
 import React from 'react';
+import { get, assign, last } from 'lodash';
 import AggSelect from './agg_select';
 import FieldSelect from './field_select';
 import AggRow from './agg_row';
 import createChangeHandler from '../lib/create_change_handler';
 import createSelectHandler from '../lib/create_select_handler';
-import createTextHandler from '../lib/create_text_handler';
+
 import {
   htmlIdGenerator,
   EuiFlexGroup,
   EuiFlexItem,
   EuiFormLabel,
-  EuiFieldText,
+  EuiFieldNumber,
   EuiFormRow,
+  EuiSpacer,
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
+import AddDeleteButtons from '../add_delete_buttons';
 
 export const PercentileRankAgg = props => {
   const { series, panel, fields } = props;
-  const defaults = { value: '' };
+  const defaults = { values: [''] };
   const model = { ...defaults, ...props.model };
 
+  const indexPattern = series.override_index_pattern && series.series_index_pattern || panel.index_pattern;
+  const htmlId = htmlIdGenerator();
+
+  const isTablePanel = panel.type === 'table';
+  const percentileRankValueRows = model.values;
+
   const handleChange = createChangeHandler(props.onChange, model);
   const handleSelectChange = createSelectHandler(handleChange);
-  const handleTextChange = createTextHandler(handleChange);
 
-  const indexPattern = series.override_index_pattern && series.series_index_pattern || panel.index_pattern;
-  const htmlId = htmlIdGenerator();
+  const handlePercentileRankValueChange = (values) => {
+    handleChange(assign({}, model, {
+      values,
+    }));
+  };
+
+  const onChangeValue = (index, event) => {
+    percentileRankValueRows[index] = get(event, 'target.value');
+
+    handlePercentileRankValueChange(percentileRankValueRows);
+  };
+
+  const onDeleteValue = index => (
+    handlePercentileRankValueChange(
+      percentileRankValueRows
+        .filter((item, currentIndex) => index !== currentIndex))
+  );
+
+  const onAddValue = () => handlePercentileRankValueChange([...percentileRankValueRows, '']);
+
+  const renderValuesRows = (value, key) => (
+    <div className="tvbAggRow__percentileRankValue" key={key}>
```

Comment:
What do you think about creating am explicit component for this? It would make code more clean. Also, you would avoid such bindings as `onChangeValue.bind(null, key)`, because you'll move handler inside the component

Reply:
@sulemanof you are partially right, but to avoid of bind I should pass model into this new component. Not sure that I want to do it.  Probably we can create a generic component for multi values select. But let's think about it next time when we need it.

Reply:
@alexwizp what is a necessity of passing model inside that component? If you need you could just pass the `handlePercentileRankValueChange` prop